src/main/java/uk/gov/laa/ccms/caab/mapper/AssessmentMapper.java

```diff
@@ -0,0 +1,497 @@
+  @Mapping(target = "completed", ignore = true)
+  AssessmentEntityDetail toAssessmentEntityDetail(ProceedingDetail proceeding);
+
+  @Mapping(target = "id", ignore = true)
+  @Mapping(target = "name", source = "opponentContext.opponent",
+      qualifiedByName = "getOpponentOpaInstanceMappingId")
+  @Mapping(target = "attributes", source = "opponentContext",
+      qualifiedByName = "opponentToAttributeList")
+  @Mapping(target = "prepopulated", constant = "true")
+  @Mapping(target = "relations", ignore = true)
+  @Mapping(target = "completed", ignore = true)
+  AssessmentEntityDetail toAssessmentEntityDetail(AssessmentOpponentMappingContext opponentContext);
+
+  /**
+   * Converts a list of opponent mapping contexts to a list of {@link AssessmentEntityDetail}
+   * objects.
+   *
+   * @param opponentContextList the list of opponent mapping contexts
+   * @return a list of assessment entity details, or {@code null} if the input list is null
+   */
+  default List<AssessmentEntityDetail> toAssessmentEntityDetailListOpponent(
+      final List<AssessmentOpponentMappingContext> opponentContextList) {
+
+    if (opponentContextList == null) {
+      return null;
+    }
+
+    final List<AssessmentEntityDetail> list = new ArrayList<>(opponentContextList.size());
+
+    for (final AssessmentOpponentMappingContext opponentContext : opponentContextList) {
+      list.add(toAssessmentEntityDetail(opponentContext));
+    }
+
+    return list;
+  }
+
+  /**
+   * Converts opponent data to a list of {@link AssessmentAttributeDetail} objects.
+   *
+   * @param opponentContext the opponent mapping context containing opponent details
+   * @return a list of assessment attribute details for the opponent
+   */
+  @Named("opponentToAttributeList")
+  default List<AssessmentAttributeDetail> opponentToAttributeList(
+      final AssessmentOpponentMappingContext opponentContext) {
+
+    final OpponentDetail opponent = opponentContext.getOpponent();
+    return List.of(
+        toAttribute(OPPONENT_DOB,
+            opponent.getDateOfBirth()),
+        toAttribute(OTHER_PARTY_ID,
+            getOpponentOpaInstanceMappingId(opponent)),
+        toAttribute(OTHER_PARTY_NAME,
+            getDisplayName(opponent,
+                opponentContext.getTitleCommonLookupValue())),
+        toAttribute(OTHER_PARTY_TYPE,
+            OpponentUtil.getOpaOpponentType(opponent.getType())),
+        toAttribute(RELATIONSHIP_TO_CASE,
+            opponent.getRelationshipToCase()),
+        toAttribute(RELATIONSHIP_TO_CLIENT,
+            opponent.getRelationshipToClient())
+    );
+  }
+
+  /**
+   * Converts global assessment data to a list of {@link AssessmentAttributeDetail} objects.
+   *
+   * @param context the assessment mapping context containing application, client, and user details
+   * @return a list of assessment attribute details
+   */
+  @Named("globalToAttributeList")
+  default List<AssessmentAttributeDetail> globalToAttributeList(
```

Comment:
Please can these be compared to the files:

com.legalservices.laa.ccms.config.MappingConfig In PUI

see - public List<Prepopulator> sharedPrepopulationMappingList() {